app/v2/controllers/requestParsers/validators/validations/JsonFormatValidation.scala

```diff
@@ -16,18 +16,48 @@
 
 package v2.controllers.requestParsers.validators.validations
 
+import play.api.Logger
 import play.api.libs.json._
-import v2.models.errors.MtdError
+import v2.models.errors.{MtdError, RuleIncorrectOrEmptyBodyError}
 
 object JsonFormatValidation {
 
-  def validate[A](data: JsValue, error: MtdError)(implicit reads: Reads[A]): List[MtdError] = {
 
-    data.validate[A] match {
-      case JsSuccess(_, _) => NoValidationErrors
-      case _               => List(error)
+  def validate[A: OFormat](data: JsValue): List[MtdError] = {
+    if (data == JsObject.empty) List(RuleIncorrectOrEmptyBodyError) else
+      data.validate[A] match {
+        case JsSuccess(body, _) => if (Json.toJson(body) == JsObject.empty) List(RuleIncorrectOrEmptyBodyError) else NoValidationErrors
+        case JsError(errors: Seq[(JsPath, Seq[JsonValidationError])]) => handleErrors(errors)
+      }
+  }
+
+
+  private def handleErrors(errors: Seq[(JsPath, Seq[JsonValidationError])]): List[MtdError] = {
+    val failures = errors.map {
+      case (path: JsPath, Seq(JsonValidationError(Seq("error.path.missing")))) => MissingMandatoryField(path)
+      case (path: JsPath, Seq(JsonValidationError(Seq(error: String)))) if error.contains("error.expected") => WrongFieldType(path)
+      case (path: JsPath, _) => OtherFailure(path)
     }
 
+    val logString = failures.groupBy(_.getClass)
+      .values.map(failure => s"${failure.head.failureReason}: " + s"${failure.map(_.fromJsPath)}")
+      .toString().dropRight(1).drop(5)
+
+    Logger.warn(s"[JsonFormatValidation][validate] - Request body failed validation with errors - $logString")
+    List(RuleIncorrectOrEmptyBodyError)
```

Comment:
You're just returning RuleIncorrectOrEmptyBodyError with no paths - what's the point of this change? Just to add logs? If the point is to return paths too, please add tests for the new functionality.

Reply:
Missed adding the path change here, now added with tests.